### PR TITLE
NF: Use isDyn/isStd more

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
This change decks so that dyn/std is used more often as function, for clarity

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Deck.java [hunk 1]
```diff
@@ -45,6 +45,6 @@ public boolean isDyn() {
     }
 
     public boolean isStd() {
-        return getInt("dyn") == Consts.DECK_STD;
+        return isStd();
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Deck.java:48]: This looks self-referential, how is this to work?

File: AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java [hunk 2]
```diff
@@ -50,4 +50,12 @@ public static boolean parseTimerOpt(JSONObject config, boolean defaultValue) {
         }
         return ret;
     }
+
+    public boolean isDyn() {
+        return getInt("dyn") == Consts.DECK_DYN;
+    }
+
+    public boolean isStd() {
+        return isStd();
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/libanki/DeckConfig.java:59]: This also looks self-referential :thinking:  ?

**Review by mikehardy — CHANGES_REQUESTED** [review]
I like the centralization of the "isStd/isDyn" logic, but the two isStd functions don't appear to actually rely on a real comparison rather they appear to refer to themselves, wouldn't that StackOverflowError ?

**Comment by Arthur-Milchior** [comment]
You're right. Added a regression test for it

**Review by mikehardy — APPROVED** [review]
All looks correct to me now :-), thanks for adding the test

**Pull request merged**